src/chainlink/factory/ChainlinkOracleFactory.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.17;
+
+import {BaseChainlinkOracleFactory} from "./BaseChainlinkOracleFactory.sol";
+import {ChainlinkOracleImpl} from "../oracle/ChainlinkOracleImpl.sol";
+
+/// @title Chainlink Oracle Factory
+/// @author 0xSplits
+/// @notice Factory for creating Chainlink Oracles
+contract ChainlinkOracleFactory is BaseChainlinkOracleFactory {
+    /// -----------------------------------------------------------------------
+    /// storage - constants & immutables
+    /// -----------------------------------------------------------------------
+
+    address public immutable ORACLE;
```

Comment:
if you name this `oracleImplementation` and give it the override keyword you don't need to redefine `oracleImplemention()` separately (having said that, might still be worth it to keep separate for CAPS usage w an immutable idk)

https://github.com/0xSplits/splits-contracts/blob/main/contracts/SplitMain.sol#L145

Reply:
yeah I did it just because its an immutable but I am okay with renaming it

Reply:
idc either way,  your call

Reply:
okay this is not possible :(
state variables can only override external functions but are baseFactory needs the function to remain public